packages/protobuf/src/private/field-wrapper.ts

```diff
@@ -45,6 +45,11 @@ export function wrapField<T extends Message<T>>(
   if (type.fieldWrapper) {
     return type.fieldWrapper.wrapField(value);
   }
+  // Last resort check if the value is indeed a Message and has the same
+  // type as the field.
+  if (isMessage<T>(value) && value.getType().typeName === type.typeName) {
+    return value;
+  }
```

Comment:
... So here, you'd instead do:

```typescript
if (type.is(value)) {
  return value;
}
```

Note: Not sure if the name `is` is a good choice, but you get the idea.

I am not a maintainer though... So :-)